tests/unit-tests/admin/test-class-sensei-onboarding.php

```diff
@@ -383,4 +382,122 @@ function() use ( $response_body ) {
 		$this->assertEquals( $extensions[0]->price, '$1.00' );
 		$this->assertEquals( $extensions[1]->price, 0 );
 	}
+
+	/**
+	 * Tests that get sensei extensions and returns with decoded prices.
+	 *
+	 * @covers Sensei_Onboarding::get_feature_with_status
+	 * @covers Sensei_Onboarding::get_sensei_extensions
+	 */
+	public function testGetSenseiExtensionsWithStatuses() {
+		// Set installing plugins
+		$installing_plugins = [
+			(object) [
+				'product_slug' => 'slug-1',
+			],
+			(object) [
+				'product_slug' => 'slug-2',
+				'error'        => 'Error message',
+			],
+		];
+		$transient          = Sensei_Plugins_Installation::INSTALLING_PLUGINS_TRANSIENT;
+		set_transient( $transient, $installing_plugins, DAY_IN_SECONDS );
+
+		// Mock fetch from senseilms.com.
+		$response_body = '{
+			"products": [
+				{ "product_slug": "slug-1", "plugin_file": "test/test.php" },
+				{ "product_slug": "slug-2", "plugin_file": "test/test.php" },
+				{ "product_slug": "slug-3", "plugin_file": "test/test-installed.php" },
+				{ "product_slug": "slug-4", "plugin_file": "test/test.php" }
+			]
+		}';
+		add_filter(
+			'pre_http_request',
+			function() use ( $response_body ) {
+				return [ 'body' => $response_body ];
+			}
+		);
+
+		// Mock active plugins.
+		$mock = $this->getMockBuilder( Sensei_Plugins_Installation::class )
+			->disableOriginalConstructor()
+			->setMethods( [ 'is_plugin_active' ] )
+			->getMock();
+
+		$mock->method( 'is_plugin_active' )
+			->will( $this->onConsecutiveCalls( false, false, true, false ) );
```

Comment:
I wanted to use the `returnValueMap` here, but for some reason, it didn't work for a method with a single parameter. I tested the same example from the documentation, but it always return `null`. 🤷‍♂️